lib/arjdbc/postgresql/adapter.rb

```diff
@@ -1012,10 +774,39 @@ module ActiveRecord::ConnectionAdapters
   remove_const(:PostgreSQLAdapter) if const_defined?(:PostgreSQLAdapter)
 
   class PostgreSQLAdapter < JdbcAdapter
+
+    # Store references to these methods so we can use them
+    # to overwrite the versions defined by
+    # ActiveRecord::ConnectionAdapters::PostgreSQL::DatabaseStatements
+    # This makes it so we can use the bulk of the core logic while
+    # keeping the actual database access defined in this adapter
+    # This is hopefully temporary until this is cleaned up enough that we
+    # can extend ActiveRecord's AbstractAdapter or PostgreSQLAdapter instead of the JdbcAdapter
+    SAVED_METHODS = ['execute', 'exec_delete', 'exec_query', 'exec_update',
+                     'select_rows', 'select_value', 'select_values',
+                     'begin_db_transaction', 'commit_db_transaction', 'rollback_db_transaction',
+                     'begin_isolated_db_transaction', 'supports_transaction_isolation?',
+                     'create_savepoint', 'rollback_to_savepoint',
+                     'release_savepoint', 'current_savepoint_name']
+    SAVED_METHOD_PREFIX = 'jdbc_adapter'
+
+    SAVED_METHODS.each do |base_name|
+      alias_method :"#{SAVED_METHOD_PREFIX}_#{base_name}", base_name
+    end
+
+    # Keep this one separate because we don't want it to be automatically redefined
+    alias_method :"#{SAVED_METHOD_PREFIX}_exec_insert", :exec_insert
+
```

Comment:
I am confused by this logic...Can you include DatebaseStatements above the definitions you want to save?